File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
import java.util.*;

class Solution {
    public int longestConsecutive(int[] nums) {

        HashSet<Integer> set = new HashSet<>();

        for(int i = 0; i < nums.length; i++){
            set.add(nums[i]);
        }

        int maxi = 0;

        Integer[] arr = set.toArray(new Integer[0]);

        for(int i = 0; i < arr.length; i++){

            int ele = arr[i];

            if(!set.contains(ele - 1)){

                int cnt = 1;

                while(set.contains(ele + 1)){
                    ele++;
                    cnt++;
                }

                maxi = Math.max(maxi, cnt);
            }
        }

        return maxi;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = 0
        seen = set(nums)
        for num in seen:
            if num-1 not in seen:
                current = num
                length = 0
                while current in seen:
                    current += 1
                    length += 1
                count = max(count,length)
        return count